src/MessagePack.UnityClient/Assets/Scripts/MessagePack/Utilities.cs

```diff
@@ -20,11 +20,17 @@ internal static class Utilities
         internal static readonly bool IsMono = Type.GetType("Mono.Runtime") is Type;
 #endif
 
+#if MESSAGEPACK_INTERNAL || DYNAMICCODEDUMPER
+        internal static SequencePool Pool { get; } = new SequencePool();
+#else
+        internal static SequencePool Pool => MessagePackSerializer.DefaultOptions.Pool;
+#endif
+
         internal delegate void GetWriterBytesAction<TArg>(ref MessagePackWriter writer, TArg argument);
 
         internal static byte[] GetWriterBytes<TArg>(TArg arg, GetWriterBytesAction<TArg> action)
         {
-            using (var sequenceRental = SequencePool.Shared.Rent())
+            using (var sequenceRental = Pool.Rent())
```

Comment:
In method `ToJsonCore` i found read of DefaultOptions. Should i change that code too, to read from parameter instead static field? 

https://github.com/neuecc/MessagePack-CSharp/blob/bc4e38fa5a16eb32c37e17172b2817ed72061a35/src/MessagePack.UnityClient/Assets/Scripts/MessagePack/MessagePackSerializer.Json.cs#L302

Reply:
I fixed some cases of this issue by passing options parameter, but I can't pass any parameter in `GetWriterBytes` because this method through `BuildFormatterToDynamicMethod` is using only in static constructor. We can use just static pool there.